cirq-core/cirq/ops/global_phase_op_test.py

```diff
@@ -279,3 +279,20 @@ def test_resolve_error(resolve_fn):
     gpt = cirq.GlobalPhaseGate(coefficient=t)
     with pytest.raises(ValueError, match='Coefficient is not unitary'):
         resolve_fn(gpt, {'t': -2})
+
+
+@pytest.mark.parametrize(
```

Comment:
Optional:  should we add a test for some of the other conditions in the code, like the ProductOfSums, that it does not return the target gate?

Reply:
done